andela/suyabay
# Create a proper authorization for Suyabay dashboard.

#### What does this PR do?
- Create a proper authorization for Suyabay dashboard.
#### Description of Task to be completed?
- Create a middleware that prevents regular users from having access to premium users' routes
- Create a middleware that prevents both regular and premium users from accessing super admin routes
- Group episodes routes' and add a middleware that prevent regular users from accessing them.
- Group users routes' and add a middleware that prevent regular and premium  users from accessing them.
- Group channels routes' and add a middleware that prevent regular and premium  users from accessing them.
#### How should this be manually tested?
- User should log in with a regular user account and visits the following routes;
-   `/dashboard`
-   `/dashboard/episode/create`
-   `/dashboard/episodes`
-   `/dashboard/edit/{id}`
-   `/dashboard/delete/{id}`

And  should see an error 401 page.
- User should log in with either  a regular or premium  user account and visits the following routes;
-   `/dashboard/channels/create`
-   `/dashboard/channels`
-   `/dashboard/channels/edit/{id}`
-   `/dashboard/channels/delete/{id}`
-   `/dashboard/users/`
-   `/dashboard/user/create`
-   `/dashboard/user/edit/{id}`
-   `/dashboard/user/delete/{id}`

And should see an error 401 page 
#### Any background context you want to provide?
#### What are the relevant pivotal tracker stories?
#### Screenshots (if appropriate)
#### Questions:

[@unicodeveloper](https://github.com/unicodeveloper)
[@andela-oadebayo](https://github.com/andela-oadebayo)
